collector/systemd_linux.go

```diff
@@ -258,6 +295,29 @@ func (c *systemdCollector) getAllUnits() ([]unit, error) {
 				nRestarts := restartsCount.Value.Value().(uint32)
 				unit.nRestarts = &nRestarts
 			}
+
+			tasksCurrentCount, err := conn.GetUnitTypeProperty(unit.Name, "Service", "TasksCurrent")
+			if err != nil {
+				log.Debugf("couldn't get unit '%s' TasksCurrent: %s", unit.Name, err)
+			} else {
+				val := tasksCurrentCount.Value.Value().(uint64)
+				// don't set if tasksCurrent if dbus reports MaxUint64
+				if val != math.MaxUint64 {
+					unit.tasksCurrent = &val
```

Comment:
don't set and don't report a metric if the TasksCurrent is MaxUint64 as seen here:
```
busctl introspect org.freedesktop.systemd1 /org/freedesktop/systemd1/unit/systemd_2drandom_2dseed_2eservice | grep Tasks
.TasksAccounting                    property  b              true                                     -
.TasksCurrent                       property  t              18446744073709551615                     -
.TasksMax                           property  t              81919
```
not sure that's the right way to handle this. maybe we should report 0 or -1 ?

Reply:
That's what I was asking myself when I saw it. MaxUint64 could be explained as infinite for TasksMax, but I cannot see a rationale for TasksCurrent = MaxUint64 :man_shrugging: 
I just would report it in this case

Reply:
Yes, let's just report `TasksCurrent` as-is. We generally want to avoid transforming metrics from another system.